### PR TITLE
Select input conditional render value

### PR DESCRIPTION
**Pull request by ChristianMoll**
Adds the `controlShouldRenderValue` prop to `<SelectField>` and `<SelectInput>`

### CODE REVIEW

**Review by Rhotimee — APPROVED** [review]
Thank you 🚀

**Review by kark — COMMENTED** [review]
Hi 👋
Thank you, it looks good to me.
Could you please add this new prop to the components' stories [here](https://github.com/commercetools/ui-kit/blob/main/packages/components/fields/select-field/src/select-field.story.js) and [here](https://github.com/commercetools/ui-kit/blob/main/packages/components/inputs/select-input/src/select-input.story.js)?

File: packages/components/fields/select-field/src/select-field.tsx [hunk 1]
```diff
@@ -113,6 +113,10 @@ export type TSelectFieldProps = {
    * [Props from React select was used](https://react-select.com/props)
    */
   components?: ReactSelectProps['components'];
+  /**
+   * Control whether the selected values should be rendered in the control
+   */
+  controlShouldRenderValue?: boolean;
```
Comment (kark) [inline packages/components/fields/select-field/src/select-field.tsx:119]: nit:
```suggestion
  controlShouldRenderValue?: ReactSelectProps['controlShouldRenderValue'];
```

Comment (kark) [inline packages/components/fields/select-field/src/select-field.tsx:119]: nit:
```suggestion
 /**
   * Control whether the selected values should be rendered in the control
   * <br>
   * [Props from React select was used](https://react-select.com/props)
   */
  controlShouldRenderValue?: ReactSelectProps['controlShouldRenderValue'];
```

**Comment by ChristianMoll** [comment]
All changes applied, thank you for the feedback ❤️

**Review by kark — APPROVED** [review]
Again, many thanks for working on this 🎉
I noticed a little regression caused by this PR, but from my side I don't want to block your work. So I would merge it and we could figure it out intrenally as a follow up. I'm wondering what you think @commercetools/shield-team-fe

**Comment by CarlosCortizasCT** [comment]
> Again, many thanks for working on this 🎉 I noticed a little regression caused by this PR, but from my side I don't want to block your work. So I would merge it and we could figure it out intrenally as a follow up. I'm wondering what you think @commercetools/shield-team-fe
> 
>  [  ](https://user-images.githubusercontent.com/49066275/228228617-d5069462-c52e-4b99-93af-7ad173be925b.gif) [ ](https://user-images.githubusercontent.com/49066275/228228617-d5069462-c52e-4b99-93af-7ad173be925b.gif)

If I got it right, to reproduce the issue you need to be using `isMulti` prop, `controlShouldRenderValue `, select several items and dynamically change the  `controlShouldRenderValue` value.
It seems to me quite an edge case so I wouldn't block this PR either.

**Pull request merged**